docs/design/mp_arktos.md

```diff
@@ -0,0 +1,145 @@
+
+### OPTION ONE: 
+Arktos adds two more optional fields into its spec, ip and prefix. This will address situations where the custom vpcID listed in Arktos network object has not been created, and Mizar needs to know the CIDR range in order to create a VPC object. The spec should show something like this: 
+```yaml
+spec:
+  type: mizar
+  vpcID: vpc-1a2b3c4d
+  ip: 10.0.0.0
+  prefix: 16
+```
+When new VPC is created and mapped with Arktos network object, the data is stored as a key/value pair in Mizar’s object store (i.e. ```dict = {Arktos_network : vpcid}```)
+
+### OPTION TWO: 
+Arktos can directly use Mizar’s VPC CRD to define its network object. As mentioned in Arktos network object overview section, Arktos network object is equivalent to VPC object in Mizar. Thus, Arktos can adapt Mizar’s VPC CRD to define Arktos network object: 
+```yaml
+apiVersion: mizar.com/v1
+kind: Vpc
+metadata:
+  name: vpc0
+spec:
+  ip: 10.0.0.0
+  prefix: 16
+  status: Init
+  dividers: 2
+```
+where status is always “Init” when creating a new VPC object, and dividers specifies the number of dividers/subnets desired. The ip and prefix fields as mentioned previous, together they define the CIDR range of the VPC. 
+
+## Analysis
+The following describes in detail on the steps involved during creation of each network object such as services and vpcs. 
+
+### The Arktos network object is associated with a vpcID that already exists:
+```yaml
+spec:
+  type: mizar
+  vpcID: vpc01
+  ip: 10.0.0.0
+  prefix: 16
+```
+* List Kubernetes cluster object
+* Find vpc object that is associated with the vpcID
+* Update object stores: map Arktos network with vpc id and stored as a key/value pair. 
+
+### The Arktos network object is associated with a vpcID that does not exist yet: 
+```yaml
+spec:
+  type: mizar
+  vpcID: vpc123f3
+  ip: 172.0.0.0
+  prefix: 16
+```
+* List Kubernetes cluster object
+* Not able to find vpc object with specified vpcID. 
+* trigger VPC creation workflow to create a new VPC with listed CIDR range. 
+* Update object store: add the new VPC object into vpc object store, and then map this Arktos network with this new vpcID.
+
+### The Arktos network object is associated with multiple services and the default network is used:
+```yaml
+spec:
+  type: mizar
+```
+* List Kubernetes cluster object
+* In case of using services, vpcID should not listed, and default network is selected
+* Waiting on customer to create services using kubectl commands: 
+```
+kubectl create -f test_service.yaml
+kubectl run pod1 --image=localhost:5000/testpod
+kubectl label pods pod1 run=example3 –overwrite
+```
+```yaml
+metadata:
+  name: test-service-3
+  annotations: 
+        service.beta.kubernetes.io/mizar-scaled-endpoint-type: “scaled-endpoint”
+  lables: 
+    run: test-service-3
+spec:
+  ports: 
+-	port: 80
+     protocol: TCP
+  selector: 
+     run: example3
+```
```

Comment:
How does arktos pass vpc/net info here?

Reply:
in short, CNI_ARGS as in cni spec.
please see [annotation as cni-ags hints](https://github.com/futurewei-cloud/arktos/pull/341) for more description.